File: tests/typesafety/typescript/exercise_shipping_start.ts
// expect: passes

function shippingCents(
  weight: number,
  distance: number,
  express: boolean,
  insured: boolean,
  fragile: boolean,
): number {
  const rate = express ? 3 : 1;
  const surcharge = (insured ? 25 : 0) + (fragile ? 40 : 0);
  return Math.trunc(weight * distance * rate) + surcharge;
}

export const quoteA = shippingCents(2.5, 120.0, true, false, true);
export const quoteB = shippingCents(120.0, 2.5, true, true, false);

File: tests/typesafety/python/exercise_shipping_start.py
# expect: passes
# title: shipping_cents takes five positional arguments, three of them bool
def shipping_cents(
    weight: float, distance: float, express: bool, insured: bool, fragile: bool
) -> int:
    rate = 3 if express else 1
    surcharge = (25 if insured else 0) + (40 if fragile else 0)
    return int(weight * distance * rate) + surcharge


quote_a = shipping_cents(2.5, 120.0, True, False, True)
quote_b = shipping_cents(120.0, 2.5, True, True, False)
